### PR TITLE
ci: scope cache to enhance speed

### PR DESCRIPTION
**Pull request by TomChv**
Resolves: #2059

Signed-off-by: Vasek - Tom C <tom.chauveau@epitech.eu>

### CODE REVIEW

**Comment by TomChv** [comment]
:warning: Do not merge it for now, I must verify that environment is working well but we can run CI in draft PR

**Comment by aluzzardi** [comment]
@TomChv Nice!

Any interesting result?

**Comment by TomChv** [comment]
There is still issues which `dagger CI / Build` that I do not understand, I think it's because go cache isn't saved locally through `#CacheDir`
But on the other side, some operation are `CACHED`.

I need more work and try to see if it's really efficient

**Comment by TomChv** [comment]
I think we should merge that one to fix our use case and see how we can find a generic way through #2140 
WDYT @aluzzardi ?

**Review by aluzzardi — APPROVED** [review]
LGTM

/cc @sipsma

File: cmd/dagger/cmd/common/common.go [hunk 1]
```diff
@@ -92,11 +92,17 @@ func NewClient(ctx context.Context) *client.Client {
 	if err != nil {
 		lg.Fatal().Err(err).Msg("unable to parse --cache-to options")
 	}
+
+	// TODO(TomChv): remove
+	fmt.Println(viper.GetStringSlice("cache-from"))
```
Comment (aluzzardi) [inline cmd/dagger/cmd/common/common.go:97]: Remove this

File: cmd/dagger/cmd/common/common.go [hunk 2]
```diff
@@ -92,11 +92,17 @@ func NewClient(ctx context.Context) *client.Client {
...
 	cacheImports, err := buildflags.ParseCacheEntry(viper.GetStringSlice("cache-from"))
 	if err != nil {
 		lg.Fatal().Err(err).Msg("unable to parse --cache-from options")
 	}
 
+	// TODO(TomChv): remove
+	fmt.Println(cacheImports)
```
Comment (aluzzardi) [inline cmd/dagger/cmd/common/common.go:104]: Remove this

File: .github/workflows/test-integration.yml [hunk 3]
```diff
@@ -29,6 +29,9 @@ on:
       - ".github/workflows/test-integration.yml"
       - "!docs/**"
 
+env:
+  DAGGER_CACHE_BASE: dagger-ci-integration
```
Comment (aluzzardi) [inline .github/workflows/test-integration.yml:33]: some workflows are setting `DAGGER_LOG_FORMAT` and some are not -- maybe standardize one or the other?

File: cmd/dagger/cmd/common/common.go (hunk 1 above)
Comment (TomChv) [reply cmd/dagger/cmd/common/common.go]: Thanks for the reminder, I updated it!

File: cmd/dagger/cmd/common/common.go (hunk 2 above)
Comment (TomChv) [reply cmd/dagger/cmd/common/common.go]: Thanks for the reminder, I updated it!

File: .github/workflows/test-integration.yml (hunk 3 above)
Comment (TomChv) [reply .github/workflows/test-integration.yml:33]: Some are set it in `helpers.bash`, I think I can take care of that in another PR

File: .github/workflows/test-integration.yml [hunk 4]
```diff
@@ -64,6 +67,18 @@ jobs:
...
 
+      - name: Sets env vars on push to main
+        run: |
+          echo "DAGGER_CACHE_TO=type=gha,mode=max,scope=${{env.DAGGER_CACHE_BASE}}-main" >> $GITHUB_ENV
+          echo "DAGGER_CACHE_FROM=type=gha,scope=${{env.DAGGER_CACHE_BASE}}-main" >> $GITHUB_ENV
+        if: ${{ github.event_name == 'push' && github.base_ref == 'main' }}
+
+      - name: Sets env vars on pull request
+        run: |
+          echo "DAGGER_CACHE_TO=type=gha,mode=max,scope=${{env.DAGGER_CACHE_BASE}}-${{github.event.number}}" >> $GITHUB_ENV
```
[the remark this answers is not in the document]
Comment (TomChv) [reply .github/workflows/test-integration.yml:78]: This is why I set `DAGGER_CACHE_BASE`, it's different in each workflows so they will not be overridden, WDYT?

[the remark this answers is not in the document]
Comment (sipsma) [reply .github/workflows/test-integration.yml:78]: Oh duh, I think I noticed that at some point but forgot on my re-read. Makes sense

**Pull request merged**